### PR TITLE
Phantom refactor yaml

### PR DESCRIPTION
**Pull request by scoates**
@ehaas for your review.
Parses either JSON or YAML.

### CODE REVIEW

File: phantompasswd.py [hunk 1]
```diff
@@ -54,15 +60,64 @@ def load_manifest(self, domain):
...
+        manifests_path = self._get_manifests_path()
+        domain = self._get_safe_domain(domain)
+
+        for content_type in [self.JSON, self.YAML]:
+            try:
+                content = open(os.path.join(manifests_path, '{}.{}'.format(domain, content_type)))
+                return self.load_data_from_content(content, content_type)
+            except IOError:
+                # file unreadable (not found)
+                pass
```
Comment (jperras) [inline phantompasswd.py:83]: Perhaps a log message entry would be good here?

**Comment by jperras** [comment]
I know you didn't ask me, but figured I'd drop in :).

Looks pretty good – only major thing I would have done differently is use the [`enum34`](https://pypi.python.org/pypi/enum34) package for enumerating the content type hints, but that's not functionally different from what you've done. Just a personal preference.

File: phantompasswd.py (hunk 1 above)
Comment (scoates) [inline phantompasswd.py:83]: It's intentionally a passthrough. It tries the `.json` file, then the `.yaml` file, and if that doesn't exist, it goes on to try `<link …` in the HTML (and other methods that we haven't documented).

**Comment by scoates** [comment]
I appreciate the opinion. (-: Even more if you want to help with this. I'd really like to make it happen this year.

I haven't used enum34. Will check it out. Originally wanted to keep this low-dependency, but Phantom helps enough to bring a bunch of stuff in.

**Comment by jperras** [comment]
Happy to help – should I submit changes to this PR, or branch off of this branch?

Are you using anything to test things out, something that I can also run/invoke to make sure I don't break anything?

**Comment by scoates** [comment]
No tests yet. It's actually incredibly difficult to test, but we could mock… well… everything.

We branched off for Phantom before Christmas. We'll probably just merge that in to master. You can PR to the `phantom` branch for now, I guess. (Or from this branch, directly in this PR if it's related.)

Let's be sure we're all (3) on the same page with what we'd like to see out of this project before we spend too much time on changes.

**Comment by jperras** [comment]
Oh I didn't mean automated tests, just anything (e.g. a site) that I can run it against that you're currently using.

**Comment by scoates** [comment]
If you want to test against your Mined VM, there's a `celobox` branch that @ehaas set up, and this:

```
python phantompasswd.py mined.fk --username celobox --oldpass password --nochange --ignore-ssl-errors
```

(you'll need to set up the user + pass)
Hopefully the other [manifests](https://github.com/scoates/celobox/tree/phantom/manifests) still work, too (you can use your own credentials, or create throwaway users).

**Comment by jperras** [comment]
Perfect, that's exactly what I was looking for. Thanks!

**Comment by scoates** [comment]
My pleasure. Excited that other people care about this project. (-:

**Comment by scoates** [comment]
:selfmerge:

**Pull request merged**